crates/storage/store/storage.rs

```diff
@@ -210,51 +218,87 @@ impl Store {
     }
 
     /// Add transaction to the pool
-    pub fn add_transaction_to_pool(
-        &self,
-        hash: H256,
-        transaction: MempoolTransaction,
-    ) -> Result<(), StoreError> {
-        self.engine.add_transaction_to_pool(hash, transaction)
-    }
-
-    /// Get a transaction from the pool
-    pub fn get_transaction_from_pool(
-        &self,
-        hash: H256,
-    ) -> Result<Option<MempoolTransaction>, StoreError> {
-        self.engine.get_transaction_from_pool(hash)
+    pub fn add_transaction_to_pool(&self, transaction: MempoolTransaction) {
+        let mut mempool = self.mempool.lock().unwrap();
+        let maybe_old_value = mempool.get_mut(&transaction.sender());
+        let sender = transaction.sender();
```

Comment:
We should call `sender()` as few as posible, as it consumes a lot of time (I think it doesn't cache it). Maybe it's better to extract it to a variable

Reply:
we can probably just cache it

Reply:
Refactored this piece of code, now we call it once. This has improved a little bit the flamegraph but it is still a bottle neck.